### PR TITLE
tie chords in note input

### PR DESCRIPTION
**Pull request by MarcSabatella**
Two things about tie entry have bothered me ever since I started using MuseScore:

- no way to tie chords while in note input mode
- tie entry in note input mode does not work after navigation (only immediately after entering a note)

This PR addresses both of these concerns.  Pressing the tie button now works even if you have navigated to the current note, and if the current note is part of a chord, the full chord is reproduced as necessary with all notes tied.

The main change regarding chords is setting the note list in note input mode to the full list of notes for the chord in the same way as when a stem is selected in normal mode.  But there are also changes to make sure we go back and add the subsequent tied notes to the same chords we generated for the first note rather than generating new chords.  And it turns out the tie-after-navigation issue is fixed by this latter change, more or less for free.

### CODE REVIEW

**Comment by MarcSabatella** [comment]
I should add, if you really want individual tied notes within a chord rather than a full tied chord, you can still do this by tying the notes outside of note input mode, or by adding the tie before adding all notes to the chord.

File: libmscore/edit.cpp [hunk 1]
```diff
@@ -1428,41 +1433,53 @@ void Score::cmdAddTie()
             }
 
       startCmd();
+      Chord* lastAddedChord = nullptr;
       foreach (Note* note, noteList) {
```
Comment (lasconic) [inline libmscore/edit.cpp]: Could be the perfect time to change the foreach to for loop?

**Comment by lasconic** [comment]
capture d ecran 2015-08-27 10 34 36

Press +
What do you expect?

Currently, you get 
capture d ecran 2015-08-27 10 35 35

**Comment by CombatCube** [comment]
The above situation would likely occur when the user desires partially tied chords. In most situations, the Note Entry tie command would be executed from a note of a chord that is before a rest or a completely untied chord, where in both situations the result is a completely tied chord. The exceptional case is when the command is executed from a partially tied chord, where partial ties should be preserved.

So for this exceptional case, it should add the source note's pitch to the chord (and only that pitch; not other untied pitches in the source chord) only if the destination chord's duration is selected; otherwise, it should replace the whole destination chord with the source chord at the specified duration as usual.

The problem is that this exception is likely to be forgotten by the user, so maybe it would be best to do a complete chord tie in all cases.

File: libmscore/edit.cpp (hunk 1 above)
Comment (MarcSabatella) [inline libmscore/edit.cpp]: Noted, can do.

**Comment by MarcSabatella** [comment]
I would love to handle that particular case as well.  The problem gets uglier if, for instance, your have some duration other than quarter note selected when you press the tie button.  If not for that concern, it would be relatively easy to handle this the way one would normally expect.

There is, BTW, an issue that discusses this case: https://musescore.org/en/node/11196

There are also other open issues that might be addressed by this PR that I'll want to look at and close manually if appropriate:

https://musescore.org/en/node/28916
https://musescore.org/en/node/58246
https://musescore.org/en/node/47596

**Comment by MarcSabatella** [comment]
Actually,, it's more complicated than that.  Consider the case of multiple ties - if after that second quarter C, there were also, say, a sixteenth note.  There is actually no single duration you could have selected that would capture that.

I *could* make it so that we check to see it there are any existing ties coming from the current chord, and if so, simply follow them and ignore the selected duration completely.  Not sure how well that would go over - it seems it would be what you want sometimes, but the current behavior might actually be preferable in other cases.

**Comment by MarcSabatella** [comment]
I think I have come up with something that works well and feels very natural.

In this situation - if there is a chord rather than a rest after the selected note - then hitting the tie button adds a note of the selected pitch to the chord (ignoring the duration from the toolbar), then ties to it.  If the chord already has a note of that pitch, it is re-used.  It makes it seem like a logical extension of the Shift+letter command for adding a note to a chord.

So, for example, starting from this:

tie-chord-1

Hitting  Shift+D adds a D to the selected chord:

tie-chord-2

Then hitting "+" adds a D to the next chord and ties it:

tie-chord-3

Then hitting "+" again ties to the existing D in the next chord:

tie-chord-4

Everything else is as it was in my original commit for this PR.

**Comment by MarcSabatella** [comment]
Just updated with another small tweak so handle case where there are already two full chords; pressing tie button now simply ties them just like when not in note input mode.

**Comment by lasconic** [comment]
Looks good. Can you squash it?

**Comment by MarcSabatella** [comment]
Done

**Pull request merged**